staging/src/k8s.io/apimachinery/pkg/api/validate/each.go

```diff
@@ -117,3 +119,57 @@ func EachMapKey[K ~string, T any](ctx context.Context, op operation.Operation, f
 	}
 	return errs
 }
+
+// UniqueValue verifies that each element of newSlice is unique.  This function
+// can only be used on types that are directly comparable. For non-comparable
+// types, use UniqueNonComparableValue.
+func UniqueValue[T comparable](_ context.Context, op operation.Operation, fldPath *field.Path, newSlice, _ []T) field.ErrorList {
+	return unique(fldPath, newSlice, func(a, b T) bool { return a == b })
+}
+
+// UniquePointer verifies that each element of newSlice is unique.  This
+// function can only be used on types that are directly comparable. For
+// non-comparable types, use UniqueNonComparablePointer.
+func UniquePointer[T comparable](_ context.Context, op operation.Operation, fldPath *field.Path, newSlice, _ []*T) field.ErrorList {
+	return unique(fldPath, newSlice, func(a, b *T) bool { return *a == *b })
+}
```

Comment:
The nil pointer derefs caught my eye.  I'm trying to reason through if we need any safeguards.  Sending a JSON request will null slice entries seems possible.  Are there other safety guards that kick in before this can run?

Reply:
I support the ban. @yongruilin is this something we can lint for?

+1 to InternalError

Reply:
I think we can do it in discovery.  It's on my list of things to look at but not critical for 1.33

Reply:
Let's hold this PR until I can offer something there.  Then we don't need the pointer variants at all.